draft-ietf-babel-information-model.md

```diff
@@ -648,42 +789,178 @@ babel-route-selected:
   is being advertised).
 
 
-# Common Objects
+## Definition of babel-hmac-obj
+
+~~~~
+  object {
+       string                rw babel-hmac-algorithm;
+       boolean               rw babel-hmac-verify-received;
+       string                rw babel-hmac-interfaces<0..*>;
+       babel-hmac-keys-obj   rw babel-hmac-keys<0..*>;
+   } babel-hmac-obj;
+~~~~
+{: artwork-align="left"}
+
+
+babel-hmac-algorithm
+: The name of the HMAC algorithm this object
+  instance uses. The value MUST be the same as one of the enumerations
+  listed in the babel-hmac-algorithms parameter.
+  An implementation MAY choose
+  to expose this parameter as read-only ("ro").
 
-## Definition of babel-credential-obj
+babel-hmac-verify-received
+: A boolean flag indicating whether HMAC hashes in received Babel messages
+  are required to be present and are verified. If this parameter is "true",
+  received messages are required to have a valid HMAC hash.
+  An implementation MAY choose
+  to expose this parameter as read-only ("ro").

```

Comment:
Similar comment here. If the implementation does not want to support changing this capability, a error message can be sent. Making it a ro is confusing at best.

Reply:
A report-only web page will not provide an error message. It just won't be possible to change the value of a parameter. Should there be some statement up front that says the option to expose "rw" parameters as read-only is only for report-only implementations (e.g., a web page). Any implementation intended for use with a fully functional management protocol MUST define these parameters as "rw".

Reply:
Yes, I would say that. 

Separately, would the default for this attribute be "false"? Because "true" means the received message will be dropped, if the hash does not match, and there would be no error message, which makes debugging the issue almost impossible.